psef/v_internal/auto_tests.py

```
@@ -418,6 +423,10 @@ def update_step_result(auto_test_id: int, result_id: int
 
     step_result.log = log
 
+    if has_attachment:
+        assert step_result.step.SUPPORTS_ATTACHMENT
```

This should raise a better exception if this is not the case.